common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java

```diff
@@ -444,6 +447,57 @@ public static void validateUpdateUser(Request userRequest) {
     }
   }
 
+  public static void externalIdsValidation(Request userRequest) {
+    if (userRequest.getRequest().containsKey(JsonKey.EXTERNAL_IDS)
+        && (null != userRequest.getRequest().get(JsonKey.EXTERNAL_IDS))) {
+      if (!(userRequest.getRequest().get(JsonKey.EXTERNAL_IDS) instanceof List)) {
+        throw new ProjectCommonException(
+            ResponseCode.dataTypeError.getErrorCode(),
+            ProjectUtil.formatMessage(
+                ResponseCode.dataTypeError.getErrorMessage(), JsonKey.EXTERNAL_IDS, JsonKey.LIST),
+            ERROR_CODE);
+      } else {
+        List<Map<String, String>> externalIds = null;
+        List<String> operationList = null;
+        try {
+          externalIds =
+              (List<Map<String, String>>) userRequest.getRequest().get(JsonKey.EXTERNAL_IDS);
+          operationList =
+              Arrays.asList(ProjectUtil.OperationType.values())
```

Comment:
Can we not avoid this: convert enum to list of string and then using list of strings in the next step? Why cant we use enum directly to check for valid values (see the method valueOf(String) in enum class)?